src/ReferenceWidget.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+	<div>
+		<div v-if="reference && hasCustomWidget" class="widget-custom">
+			<div ref="customWidget" />
+		</div>
+
+		<a v-else-if="!noAccess && reference && reference.openGraphObject && !hasCustomWidget" :href="reference.openGraphObject.link" class="widget-default">
+			<div v-if="reference.openGraphObject.thumb" class="widget-default--image" :style="{ 'backgroundImage': 'url(' + reference.openGraphObject.thumb + ')' }" />
+			<div class="widget-default--details">
+				<p class="widget-default--title">{{ reference.openGraphObject.name }}</p>
+				<p class="widget-default--description">{{ reference.openGraphObject.description }}</p>
+				<p class="widget-default--link">{{ reference.openGraphObject.link }}</p>
+			</div>
+		</a>
+	</div>
+</template>
+<script>
+import { renderWidget, isWidgetRegistered } from './widgets.js'
+
+export default {
+	name: 'ReferenceWidget',
+	props: {
+		reference: {
+			type: Object,
+			required: true
+		}
+	},
+	data() {
+		return {}
+	},
+	computed: {
+		hasCustomWidget() {
+			return isWidgetRegistered(this.reference.richObjectType)
+		},
+		noAccess() {
+			return this.reference && !this.reference.accessible
+		}
+	},
+	mounted() {
+		this.renderWidget()
+	},
+	methods: {
+		renderWidget() {
+			if (this.$refs.customWidget) {
+				this.$refs.customWidget.innerHTML = ''
+			}
+			if (this?.reference?.richObjectType === 'open-graph') {
+				return
+			}
+			this.$nextTick(() => {
+				// Waiting for the ref to become available
+				renderWidget(this.$refs.customWidget, this.reference)
```

Comment:
This can be dangerous.

`customWidget` is a DOM element, and can be replaced any time.

Using this as argument, we can create a memory leak, miss references or reactivity.

Reply:
Added a callback handler for the destruction, so that registering widgets can take care of the cleanup of their vue instances.